### DISCUSSION TITLE
Welcome to ng-alain Discussions!

### DISCUSSION THREAD

**Discussion by cipchk** [ng-alain/ng-alain#1900] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we build together 💪.

----------------

## 👋欢迎来到讨论区
   我们正在使用“讨论区”作为与社区其他成员联系的地方。 

   你可以在这里：
   * 问您想知道的问题。
   * 分享意见。
   * 与其他社区成员互动。

**Comment by yanyan920** [comment]
'reuse-tab' is not a known element

**Reply by cipchk** [reply]
Pls import `ReuseTabModule` in `shared-delon.module.ts`.

**Reply by yanyan920** [reply]
好滴我试试看，谢谢！

**Reply by yanyan920** [reply]
我shared-delon.module.ts里有导入ReuseTabModule，可是怎么还是不行？

**Reply by cipchk** [reply]
> 我shared-delon.module.ts里有导入ReuseTabModule，可是怎么还是不行？

见：https://github.com/ng-alain/ng-alain/blob/master/src/app/global-config.module.ts#L29

**Comment by sy-yan** [comment]
I Use st in my project。but when i use st.addRow() ,it add the column's name as a prefix.for example ,i want "a" in clomn "b"，but i got "ba".
I tried the example code in ng-alain.com,but it got a wrong result either.

**Reply by sy-yan** [reply]
import { Component } from '@angular/core';

import { STChange, STClickRowClassNameType, STColumn } from '@delon/abc/st';

@Component({
  selector: 'components-st-row-op',
  template: ` <button nz-button (click)="st.addRow({ id: 1000, email: 'add@email.com', phone: '123' })"> addRow</button>
    <button nz-button (click)="st.removeRow(0)"> removeRow index: 0 </button>
    <button nz-button (click)="st.setRow(0, { className: 'text-success' })"> Via setRow method </button>
    <st
      #st
      [data]="url"
      [req]="{ params: params }"
      [columns]="columns"
      (change)="_click($event)"
      [clickRowClassName]="clickRowClassName"
    ></st>`
})
export class ApplyComponent {
  url = [];
  params = { a: 1, b: 2 };
  clickRowClassName: STClickRowClassNameType = { exclusive: true, fn: () => 'text-error' };
  columns: STColumn[] = [
    { title: '编号', index: 'id' },
    { title: '邮箱', index: 'email' },
    { title: '电话', index: 'phone' }
  ];

  _click(e: STChange): void {
    console.log('click', e);
  }
}
